### ISSUE TITLE
Single State: Allow users to set a default when no data exists

### ISSUE THREAD

**Issue by BondAnthony** [influxdata/influxdb#19948]
__Proposal:__
Users should be able to set a default value when no data exists. Optionally users should have the ability to convert numeric values or null to a user defined string. Grafana solves this problem by allowing users to set values to text. 

__Current behavior:__
"No Results" text is displayed in the single state panel.

__Desired behavior:__
The ability to hardcode a single value of 0, Healthy, Alive, N/A, No data, etc when my queries return zero data.

__Use case:__
When I have a dashboard full of cells and healthy state is no data, it would be nice to show 0, Healthy, etc instead of `No Results`. The no results makes me second guess if the query is correct or if my system is operating as expected. A lot of services may not expose a specific measurement when everything is operating as expected. You can imagine a dashboard that has multiple cells with "No Results" doesn't look very useful.

**Comment by timhallinflux** [comment]
@sanderson essentially if the result is null, you want to return a value.  I believe you can implement a workaround today in Flux....

**Comment by BondAnthony** [comment]
I would be interested in the flux workaround. I was trying a couple things but no luck.

**Comment by sanderson** [comment]
So there are two situations to consider here:

1. **The query returns results, but the latest value is _null_.**

    This is simple enough to fix. Just include `fill()` at the end of the query and pass in the column name and default value.

    ```js
    from(bucket: "example-bucket")
      |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
      |> filter(fn: (r) => ...some filters... )
      |> fill(column: "_value", value: "Default value")
    ```

2. **There are no results. The query simply doesn't return anything.**

    This one is a little more challenging and I'd have to think a little more about how you'd do it in Flux, if it's even currently possible. You'd essentially have to check the length of the results and provide an alternate stream of tables if that length is 0.

**Comment by BondAnthony** [comment]
Right, fill! I can definitely use that when my results come back null. Thank you @sanderson
